### DISCUSSION TITLE
Error finding snapshot file in packaged executable

### DISCUSSION THREAD

**Discussion by orangelckc** [yao-pkg/pkg#41] [Q&A]
Thanks for your continued work on the project. I am currently facing an issue with the snapshot system and was wondering if someone could assist me.

Here is the code snippet that I am working with:

```js
const wakePath = join(__dirname, 'assets/naonao.table')
```

I have included the `.table` file as an asset to be packaged into a single executable file. In order to check if the file was correctly included, I printed the `md5` of the file which matched the original file. 

However, when I pass the snapshot path to the function, it always shows an error when trying to find this file. 

![image](https://github.com/yao-pkg/pkg/assets/48410934/145c9e8b-60bc-40ba-8e2a-411ed515efa6)

Could someone please help me with this issue?

**Comment by robertsLando** [comment]
Hi @orangelckc. I suggest you to try packaging the application using `-d` to see if there are errors while packaging that file into the executable. It is possible that for some reason that file fails to be read and you should firstly copy it to a temporary folder before opening. This is exact the same we have to do with `.node` native addons: https://github.com/yao-pkg/pkg/blob/main/prelude/bootstrap.js#L2199
